### PR TITLE
✨RabbitMQ: Add RPC decorator for server side

### PR DESCRIPTION
**Pull request by sanderegg**
## What do these changes do?
- [x] provides a ```@expose``` decorator to use on methods to expose as RPC methods

NOTE: this provides a convenient way to expose RPC functions. NOT yet a convenient way to provide the client side. this might come later. baby steps as someone much smarter than me said.

This PR allows to reduce the complexity of https://github.com/ITISFoundation/osparc-simcore/pull/4602

## Related issue/s
- pre-requisite of https://github.com/ITISFoundation/osparc-simcore/pull/4602

### CODE REVIEW

File: packages/service-library/tests/rabbitmq/test_rabbitmq_rpc_router.py [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+
+
+@pytest.fixture
+def rabbitmq_rpc_client(
+    rabbitmq_client: Callable[[str], RabbitMQClient]
+) -> Callable[[str], Awaitable[RabbitMQClient]]:
+    async def _creator(name: str) -> RabbitMQClient:
+        rpc_client = rabbitmq_client(name)
+        assert rpc_client
+        await rpc_client.rpc_initialize()
```
Comment (pcrespov) [inline packages/service-library/tests/rabbitmq/test_rabbitmq_rpc_router.py:25]: @GitHK @sanderegg This is a good example for a question i want to ask. here i want a rpc client (see? you even write that in the fixture!)

 Then Why don't we have a client specific for rpc? 
Can you tell me a use-case in which a `rabbitmq_client` uses both rpc and pub/sub interfaces at the same time??

**Review by pcrespov — COMMENTED** [review]
This is a nice step. The routers are great addition to simplify usage. I added a note to the general concept of "rabbit_rpc" . I think the test shows clearly things that can be improved in subsequent PRs.

File: packages/service-library/tests/rabbitmq/test_rabbitmq_rpc_router.py [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+def router_namespace(faker: Faker) -> RPCNamespace:
+    return faker.pystr()
+
+
+async def test_exposed_methods(
+    rabbitmq_rpc_client: Callable[[str], Awaitable[RabbitMQClient]],
+    router_namespace: RPCNamespace,
+    cleanup_check_rabbitmq_server_has_no_errors: None,
+):
+    rpc_client = await rabbitmq_rpc_client("client")
```
Comment (pcrespov) [inline packages/service-library/tests/rabbitmq/test_rabbitmq_rpc_router.py:62]: Same here. I think the design should limit functions exposed for `rpc_server` and those by `rpc_client`

**Review by pcrespov — APPROVED** [review]

File: packages/service-library/tests/rabbitmq/test_rabbitmq_rpc_router.py (hunk 1 above)
Comment (sanderegg) [reply packages/service-library/tests/rabbitmq/test_rabbitmq_rpc_router.py:25]: I agree. I will have a look now if this is easily doable

Comment (sanderegg) [reply packages/service-library/tests/rabbitmq/test_rabbitmq_rpc_router.py:25]: but in a separate PR

Comment (sanderegg) [reply packages/service-library/tests/rabbitmq/test_rabbitmq_rpc_router.py:25]: as for the use-case. I think it does not exist

Comment (GitHK) [reply packages/service-library/tests/rabbitmq/test_rabbitmq_rpc_router.py:25]: I think the RPC is an extension on top of rabbitmq. What will happen when we have such a use case? Will I have different clients with different connections to rabbitmq? Is that what we want anyway?

File: packages/service-library/tests/rabbitmq/test_rabbitmq_rpc_router.py [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+    rpc_server = await rabbitmq_rpc_client("server")
+
+    a_arg = "The A-Team"
+    a_kwargs = "What about McGiver?"
+    a_specific_kwarg = "Yeah, it was actually good, too!"
```
Comment (GitHK) [inline packages/service-library/tests/rabbitmq/test_rabbitmq_rpc_router.py:67]: so these are always passed in, they will be something like the `app` or some other objects that might be required?

**Review by GitHK — APPROVED** [review]

File: packages/service-library/tests/rabbitmq/test_rabbitmq_rpc_router.py (hunk 1 above)
Comment (sanderegg) [reply packages/service-library/tests/rabbitmq/test_rabbitmq_rpc_router.py:25]: yes

File: packages/service-library/tests/rabbitmq/test_rabbitmq_rpc_router.py (hunk 3 above)
Comment (sanderegg) [reply packages/service-library/tests/rabbitmq/test_rabbitmq_rpc_router.py:67]: exactly

**Review by matusdrobuliak66 — APPROVED** [review]
:+1:

**Pull request merged**